submissions/evgeniy24/friends_app/app.js

```diff
@@ -0,0 +1,234 @@
+    moreWrap.classList.add('moreWrap');
+    friendPhotoWrap.classList.add('cardWrap__photo-wrap');
+    friendPhoto.classList.add('cardWrap__photo-img');
+    friendsName.classList.add('mainInfo__friendsName');
+    friendAge.classList.add('mainInfo__age');
+    friendEmail.classList.add('moreWrap__email');
+    friendPhone.classList.add('moreWrap__phone');
+    friendCity.classList.add('mainInfo__city');
+    
+    friendsName.textContent = `${friend.name.first} ${friend.name.last}`;
+    friendAge.textContent = `Age: ${friend.dob.age}`;
+    friendEmail.textContent = `email: ${friend.email}`;
+    friendPhone.textContent = `phone: ${friend.phone}`;
+    friendCity.textContent = `City: ${friend.location.city}`;
+
+    friendPhoto.setAttribute('src', friend.picture.medium);
+    friendsName.setAttribute('href', `https://www.facebook.com/search/top?q=${friend.name.first}%20${friend.name.last}`);
+    friendsName.setAttribute('target', '_blank');
+    friendEmail.setAttribute('href', `mailto:${friend.email}`);
+    friendPhone.setAttribute('href', `tel:>${friend.phone}`);
+    friendCity.setAttribute('href', `https://www.google.com/maps/place/${friend.location.city}$`);
+    friendCity.setAttribute('target', '_blank');
+
+    moreWrap.append(friendPhone, friendEmail);
+    mainInfo.append(friendsName, friendAge, friendCity);
+    friendPhotoWrap.append(friendPhoto);
+    cardWrap.append(friendPhotoWrap, mainInfo, moreWrap);
+    FRIENDS_LIST_WRAP.append(cardWrap);
+}
+
+//find a friend by the name ------------------------------------
+SEARCH_FIELD.addEventListener('input', function() {
+    let searchName = SEARCH_FIELD.value.toLowerCase();
+    if (searchName !== '') {
+            searchResultList = friendsList.filter(function(friend) {
+            return friend.name.first.toLowerCase().includes(searchName) 
+        })
+        return renderFriendsList(searchResultList);
+    }  else {
+        renderFriendsList(friendsList);
+    }
+})
+
+//sorting List of friends --------------------------------------------
+SORTING_MENU.addEventListener('click', function(event) {
+    if (searchResultList) {
+        sortedList = searchResultList;
+    } else if (filteredByGenderList) {
+        sortedList = filteredByGenderList;
+    } else {
+        sortedList = [...friendsList];
+    }
+
+    const sortByName = function(a, b) {
+        const nameA = a.name.first.toUpperCase();
+        const nameB = b.name.first.toUpperCase();
+        return nameA < nameB ? -1 : nameA > nameB ? 1 : 0;
+    }
+
+    const sortByAge = function(a, b) {
+        return (a.dob.age - b.dob.age);
+    }
+    
+    if (event.target === SORT_BY_NAME_AZ) {
+        sortedList.sort((a, b) => sortByName(a, b));
+        renderFriendsList(sortedList);
+
+    } else if (event.target === SORT_BY_NAME_ZA) {
+        sortedList.sort((a, b) => sortByName(b, a));
+        renderFriendsList(sortedList);
+
+    } else if (event.target === SORT_BY_AGE_YO) {
```

Comment:
what the meaning of `YO`? 🤔 It's a ninja code ;)

Reply:
Young Old and vice versa!)
I will rename it

Reply:
Renamed! See a new commit!